### DISCUSSION TITLE
Program Crash and German Patch 1.5

### DISCUSSION THREAD

**Discussion by DHFan45** [THGSCST/Vices-and-Sins-MOD#12] [General]
Hello

1)
Every time I try to **_open_** one of the original files of the game ".tyl or .spr" with the program "SPRTYL2PNG" it closes (crashes) without any further information.

What information do you need to troubleshoot?

Tested with Antivirus Software on and off
Gangsters GOG Version
Win 10 x64

2)
After extracting and editing the ".xtx" files, how can I convert them back to the ".xtx" again?

**Comment by THGSCST** [comment]
Hi,

Thanks for the report. I checked and you are right, a `System.InvalidCastException` occurs. Strange that this has never happened before, maybe something has changed in .NET, who knows.

Anyway I updated the code and compiled it again. You can use version 1.1.1: https://github.com/THGSCST/Vices-and-Sins-MOD/releases/tag/r1.1.1

About XTX files, converting back is easy. I didn't implement it because the game will load TXT with priority above XTX, so if you have the 2 file extensions in the Data folder, then only the TXT will be loaded.

**Comment by DHFan45** [comment]
Thank you very much it works.

---
I would like to mention that there is a perfect patch for gangsters to make the game run on modern computers.
It also uses the "DxWrapper" (from elishacloud) but also fixes the problem that no music is played in the game. (Turn on music in the game options)

Unfortunately the patch is only available in german.
https://www.compiware-forum.de/downloads/file/782-gangsters-organisiertes-verbrechen-german-patch/?highlight=gangsters#overview


The installer of the patch can be easily unpacked with "UniExtract 2.0.0 RC 3".
https://github.com/Bioruebe/UniExtract2

Sadly, it is very difficult for me to analyze which files are necessary and which are only for the German language. Perhaps the "Gangsters.exe" and "GOGLauncher.exe" was also modified?

Would it be possible for you to analyze the patch and recompile it (e.g. as a RarArchive) so that it is suitable for the English GOG version of the game and its players?
I think alot of people in the GOG Forum would be very happy.

**Comment by THGSCST** [comment]
This is a very interesting patch, I had never seen these 2 new scenarios. About the music, this patch uses new DLLs and the tracks have been encoded in Ogg Vorbis (high quality). Soon, when I can, I will do a binary comparison and see the changes.

In GOG version, the music bug could be a version or configuration issue. First, the sound volume in windows needs to be turned up. Second, the game starts with the music disabled by default, it is necessary to disable the Ambience Music by clicking on the box until the red border comes out:

![image](https://github.com/THGSCST/Vices-and-Sins-MOD/assets/18038961/d9d5611c-2790-4d37-b957-d53edb5edc55)

**Comment by DHFan45** [comment]
Thank you
Related to playing the music: In the GOG version this works without problems. I had this wrong in memory, only with the retail (CD) version i had problems. Like, for example, the endless playback of the music does not work.

**Comment by DHFan45** [comment]
Hello, do you have any update if it is possible to convert the german patch for the english version?

**Comment by THGSCST** [comment]
Hello. You have two options:
1. You can convert this DE version to EN. You will need to replace the `gangsters.exe` executable, DATA and HELP folders with the EN version. Perhaps one or another graphic element will be in German, it will not affect the gameplay.
2. You can use the music patch only. There is no change in the exe, the hook occurs in `dxwplay.dll`, which calls the other Ogg/Vorbis DLLs that use the ogg tunes in MUSIC folder.

I did it both ways. Regardless of which option you choose, I recommend that you use the newest dxwrapper from elishacloud which can be obtained here: https://github.com/elishacloud/dxwrapper/wiki/Gangsters-Organized-Crime

**Comment by DHFan45** [comment]
Hello thank you very mutch.
Can you please tell me which game file (or combination) is behind each of the following changes?

- Recruitment has been made easier. --> ?
- The amount of business in a city has been redistributed.  -->
- Police and FBI presence has been reduced. -->  Constants.xtx ?
- Some buildings have a new name and function. -->
- New hood generation template. -->
- Ideal fronts for ilegal business has been rethought. -->  Business Suspicion.xtx ?


How can you read the tables at all? everything is totally shifted with Windows Editor or Notepad++
what program do you use?
![ghfdg](https://github.com/THGSCST/Vices-and-Sins-MOD/assets/123426586/8f6aecbd-c733-4d93-8436-b50f4073b950)

**Comment by THGSCST** [comment]
Hi.

> Recruitment has been made easier. --> ?
The amount of business in a city has been redistributed. -->
Some buildings have a new name and function. -->

In `Economics.txt`, you can modify these parameters to change the frequency of business in map generation: `City Capacity`, `City Cap Min`, `City Cap Max` and `Numbers Present`. Recruitment of new hoods is done in some specific business (Poolhall and Gym) whose frequency I have doubled, so recruitment is now easier that before. You can change some building names and their functions by changing other parameters in this file, but be aware that if something goes wrong the game will crash, without a message. So I recommend gradual changes and save each time, documenting the effectiveness of each change.

> Police and FBI presence has been reduced. --> Constants.xtx ?
> Ideal fronts for ilegal business has been rethought. --> Business Suspicion.xtx ?

Yes.

> New hood generation template. -->

I have redone all the `Character Generation.txt` file, its 100% modified. I also include some comments where I describe each skill.

>How can you read the tables at all? everything is totally shifted with Windows Editor or Notepad++
what program do you use?

This file format is old and is called Tab-separated values (TSV). It is important to be careful with newlines and tab chars. I recommend that you use Excel to view and make changes, then press Ctrl+A to select all press Ctrl+C to copy, open any text editor (can be Notepad++ that you already use) press Ctrl+V to paste and save using the text editor.

I intend in the future to completely modify all XTX files, as I did with Character Generation.txt and include comments that can document what each parameter does. As soon as I have time I will publish a new version of this mod.

**Comment by DHFan45** [comment]
Hello thanks for all the help.

Excel requires three steps when importing the text files.
Are the default settings by excel ok or do i have to adjust something?

![1](https://github.com/THGSCST/Vices-and-Sins-MOD/assets/123426586/0ae0bbb7-df0c-4c29-a9c2-8ac6ce7ff5f8)
(sorry i only have excel in german but the settings should look the same in english)

**Comment by THGSCST** [comment]
> Excel requires three steps when importing the text files. Are the default settings by excel ok or do i have to adjust something?

After thinking more, I came to the conclusion that the easiest and most convenient way to edit these files would be to use Excel as if it were a common spreadsheet, taking advantage of all the features such as formulas and formatting. After that, the final step would be to convert to TXT.

For this I created a new tool called XLSX2Data (https://github.com/THGSCST/Vices-and-Sins-MOD/commit/637089b7b840c6f6dbbd9b67e47bf9f96d507de7). Basically, you edit the Data.xlsx spreadsheet while the XLS2Data tool is running, TXT files are automatically generated. https://github.com/THGSCST/Vices-and-Sins-MOD/releases/tag/r1.2b

**Comment by DHFan45** [comment]
Thank you and excuse my very long absence

But still i dont understand:

Excel requires three steps when **_importing_** a XTX file. (see above)  I expressed myself incorrectly)
Are the default settings by excel ok or do i have to adjust something?


But if i understand it correctly every Data that i edit via your Data.xlsx  spreadsheet will do the job, so i do not have to work with the XTX files anymore?

**Comment by THGSCST** [comment]
> Are the default settings by excel ok or do i have to adjust something?

Your settings are correct, except for the Japanese encoding at Step 1. The right encoding is `1252`.

>But if i understand it correctly every Data that i edit via your Data.xlsx spreadsheet will do the job, so i do not have to work with the XTX files anymore?

Yes, exactly! Working with separate files was becoming too complicated and error-prone. With this tool, everything is much easier—just edit `Data.xlsx` while `XLSX2Data` is running.

**Comment by DHFan45** [comment]
This solution is ingenious.

I have a please.

would it be possible to provide the “Data.xlsx spreadsheet” in two versions:

A) with the original and unmodified values of the patched game (1.4 / 1.5)

B ) your modified version

------------------------------------------------------------------------------

The second request  relates to your following answer:

https://github.com/THGSCST/Vices-and-Sins-MOD/issues/7#issuecomment-1721129434

Unfortunately it is not so easy to use the "music patch" for the english version.
I have compared the files and changes that the german patch makes.

The patch removes the “patch.dll” from the english GoG version and also replaces the “gangsters.exe"

The “patch.dll” must be removed for the “music patch” to work. 
->> if you install the "music patch" and remove the GoG mp3 files there is no more music = conclusion: the "music patch" is ignored as long as the “patch.dll” is present.

However, the English GoG “gangsters.exe” seems to be attached to the “patch.dll”.

since we do not want to use the german “gangsters.exe” and the english “gangsters.exe” does not work without the “patch.dll”, it is not possible to use the “music patch” with the english version of the game.

![Image](https://github.com/user-attachments/assets/2de4047b-5d58-435e-8da0-a0dc0d31df89)

------------------------------------------------------------------------------
Note:
I have contacted the german patch creator via their forum.
The patch 1.5 were only integrated into a German retail version and never released as a download.

There is no record of what changes were made in version 1.5. it is possible that the most important change is the two additional scenarios.

A readme attached in german only contains this information:

GANGSTERS - DAS ORGANISIERTE VERBRECHEN
Version 1.5
Readme-Datei
18.08.99
This version of “Gangsters: Organized Crime” contains patch 4.0 and all other patches released by Hothouse Creations Limited up to the date shown above. Therefore, do not install any patches with an earlier date.

[README.pdf](https://github.com/user-attachments/files/18725054/README.pdf)

**Comment by THGSCST** [comment]
>would it be possible to provide the “Data.xlsx spreadsheet” in two versions:
A) with the original and unmodified values of the patched game (1.4 / 1.5)

I no longer have the original file. You can try starting as I did. Attached is an example file, [Data.xlsx](https://github.com/user-attachments/files/18731467/Data.xlsx). 
Note that there are two sheets, and the sheet name must match the file name. In the "Economics.txt" sheet, I simply copied the TXT directly into Excel. For the XLSX2Data tool to work, you need to add the exact text "### End of File ###" at the end.

>Unfortunately it is not so easy to use the "music patch" for the english version.

This German version is a retail CD (Non-GoG). When you asked about converting from German to English, I suggested replacing files in the folders and the EXE, but I wasn't referring to mixing DLLs from the CD and GoG versions. I managed to get the music working, but it was an English retail CD version.

**Comment by THGSCST** [comment]
>There is no record of what changes were made in version 1.5. it is possible that the most important change is the two additional scenarios.

Looking at the binary, it is clear that this version 1.5 was compiled after 1.4. In the PE file header information, we have the dates April 28, 1999, for version 1.4 and August 18, 1999, for version 1.5. The no-CD crack was inserted later, in 2021, I believe. I am attaching the binaries in case anyone wants to conduct a more detailed analysis: [Gangsters_DE_14_15 (retail CD, no-GoG).zip](https://github.com/user-attachments/files/18750706/Gangsters_DE_14_15.retail.CD.no-GoG.zip) (I extracted the version 1.4 from the official patch.)
